clients/client/src/main/java/org/projectnessie/client/http/HttpRequest.java

```diff
@@ -140,8 +142,21 @@ private HttpResponse executeRequest(Method method, Object body) throws HttpClien
           }
         }
         con.connect();
-        con.getResponseCode(); // call to ensure http request is complete
-
+        con.getResponseCode();
+        if (con.getResponseCode() == 308 || con.getResponseCode() == 307) {
+          String newLocation = con.getHeaderField("Location");
+          return new HttpRequest(
```

Comment:
Redirects should be guarded with a "max number of redirects" (which is what `HttpURLConnection` does).
In this case, for HTTP/308, it's probably sufficient to only handle the HTTP/308 case once, because a http->https redirect should only happen once.
Proposal:
A `do {} while (false)` loop around the outermost `try`-block.
In the `if` handling the 308 check if a HTTP/308 was already handled. If not, then update the uri to the `Location` header. If it was already handled, then error out.
If the `Location` header's missing, error out immediately (this is what all client implementation do).
This should guard the Nessie http client from an endless loop caused by a misconfigured server.